native/src/main/java/io/ballerina/persist/components/Function.java

```diff
@@ -0,0 +1,155 @@
+        }
+    }
+
+    private FunctionSignatureNode getFunctionSignature() {
+        return NodeFactory.createFunctionSignatureNode(
+                SyntaxTreeConstants.SYNTAX_TREE_OPEN_PAREN,
+                AbstractNodeFactory.createSeparatedNodeList(parameters),
+                SyntaxTreeConstants.SYNTAX_TREE_CLOSE_PAREN,
+                returnTypeDescriptorNode
+        );
+    }
+
+    public void addRequiredParameter(Node typeName, String name) {
+        if (parameters.size() > 0) {
+            parameters.add(SyntaxTreeConstants.SYNTAX_TREE_COMMA);
+        }
+        NodeList<AnnotationNode> annotations = NodeFactory.createEmptyNodeList();
+        parameters.add(
+                NodeFactory.createRequiredParameterNode(
+                        annotations,
+                        typeName,
+                        AbstractNodeFactory.createIdentifierToken(name)
+                )
+        );
+    }
+
+    public void addRequiredParameterWithDefault(Node typeName, String name) {
+        if (parameters.size() > 0) {
+            parameters.add(SyntaxTreeConstants.SYNTAX_TREE_COMMA);
+        }
+        NodeList<AnnotationNode> annotations = NodeFactory.createEmptyNodeList();
+        parameters.add(
+                NodeFactory.createDefaultableParameterNode(
+                        annotations,
+                        typeName,
+                        AbstractNodeFactory.createIdentifierToken(name),
+                        SyntaxTreeConstants.SYNTAX_TREE_EQUAL,
+                        NodeFactory.createListConstructorExpressionNode(
+                                SyntaxTreeConstants.SYNTAX_TREE_OPEN_PAREN, AbstractNodeFactory
+                                        .createSeparatedNodeList(NodeFactory.createBasicLiteralNode(
+                                                SyntaxKind.NIL_LITERAL,
+                                                AbstractNodeFactory.createLiteralValueToken(SyntaxKind.NIL_LITERAL,
+                                                        "", NodeFactory.createEmptyMinutiaeList(),
+                                                        NodeFactory.createEmptyMinutiaeList())))
+                                , SyntaxTreeConstants.SYNTAX_TREE_CLOSE_PAREN)
+                        )
+        );
+    }
+
+    public void addReturns(TypeDescriptorNode node) {
+        returnTypeDescriptorNode = getReturnTypeDescriptorNode(node);
+    }
+
+    private FunctionBodyNode getFunctionBody() {
+        return NodeFactory.createFunctionBodyBlockNode(
+                SyntaxTreeConstants.SYNTAX_TREE_OPEN_BRACE,
+                null,
+                statements,
+                SyntaxTreeConstants.SYNTAX_TREE_CLOSE_BRACE
+        );
+    }
+
+    public void addStatement(StatementNode node) {
+        statements = statements.add(node);
+    }
+
+    public void addIfElseStatement(IfElseStatementNode node) {
+        statements = statements.add(node);
+    }
+
+}
+
```

Comment:
```suggestion
```

Reply:
Updated in [d6a53d3](https://github.com/ballerina-platform/persist-tools/pull/15/commits/d6a53d3f948c8f076acb3cc1f2730258a998cf40)